airflow/models/xcom.py

```diff
@@ -364,6 +372,12 @@ def orm_deserialize_value(self) -> Any:
         return BaseXCom.deserialize_value(self)
 
 
+def serialize_value_shim(**kwargs):
+    signature = inspect_function_arguments(XCom.serialize_value)
+    kwargs = {k: v for k, v in kwargs.items() if k in signature.bound_arguments}
+    return XCom.serialize_value(**kwargs)
```

Comment:
Worth a docstring explaining why this is needed.

Reply:
Also I wonder if it’d be a good idea to emit a warning telling the XCom backend to upgrade to the new arguments.

Reply:
> Worth a docstring explaining why this is needed.

yeah true i have explanation in test i can move or copy here

> Also I wonder if it’d be a good idea to emit a warning telling the XCom backend to upgrade to the new arguments.

yeah if we're going deprecation route that makes sense

also, i've been very permissive here allowing any combination of kwargs --- do you think i should stick with that or just allow 2 options: (1)  value only and (2) all params in `set`?

Reply:
Being permissive is OK IMO, no need to complicate things too much for things that would likely not going to work in the end anyway.

Putting the exaplaination in the implementation is much better than in tests since it’s easy to find the implementation when you can’t understand a test, but the other way around is much more difficult.

Reply:
Very much agree with that - additionally, it's good to add more explanation in the commit message - so that it could be find in the history as well.

Reply:
I learned to provide all the "why's" and context in the commit messages mostly for my future self, as I realized that I can't remember some of the why's few months after I made a change. It saved me quite a number of times.

Reply:
yeah totally, and truthfully i intended to add a docstring there but was just taking a couple minutes to refactor last night... probably should have been marked as a draft...  i think flake8 fails when you omit docstring too but i disable locally sometimes when i don't want docker running...  i just meant to indicate "yup i have some explanation ready to go"

anyway docstring is there now.  i tweaked the approach slightly to iterate over `bound_arguments` instead of keys ... seems a little cleaner this way.  

one potential optimization is we could sort of "cache" the signature inspection so we don't have to do it every call.  or even cache the shim function so it doesn't have to do any logic... but both operations are cheap and probably it's not worth the clutter.

lmkt

thanks